src/bp/ui-admin/src/Pages/Workspace/Bots/index.tsx

```diff
@@ -1,54 +1,55 @@
+import {
+  Alignment,
+  Button,
+  ButtonGroup,
+  Callout,
+  Intent,
+  Popover,
+  PopoverInteractionKind,
+  Position
+} from '@blueprintjs/core'
+import _ from 'lodash'
 import React, { Component, Fragment } from 'react'
-
 import { connect } from 'react-redux'
-import { Row, Col, Alert } from 'reactstrap'
-
-import _ from 'lodash'
+import { RouteComponentProps } from 'react-router'
+import { Alert, Col, Row } from 'reactstrap'
+import { toastSuccess } from '~/utils/toaster'
+import { Downloader } from '~/Pages/Components/Downloader'
 
+import api from '../../../api'
 import { fetchBots } from '../../../reducers/bots'
-import { fetchPermissions } from '../../../reducers/user'
 import { fetchLicensing } from '../../../reducers/license'
-
-import SectionLayout from '../../Layouts/Section'
+import AccessControl from '../../../App/AccessControl'
 import LoadingSection from '../../Components/LoadingSection'
+import SectionLayout from '../../Layouts/Section'
 
-import api from '../../../api'
-import { AccessControl } from '../../../App/AccessControl'
+import BotItemCompact from './BotItemCompact'
+import BotItemPipeline from './BotItemPipeline'
 import CreateBotModal from './CreateBotModal'
 import ImportBotModal from './ImportBotModal'
-import BotItemPipeline from './BotItemPipeline'
-import BotItemCompact from './BotItemCompact'
 import RollbackBotModal from './RollbackBotModal'
-import { toast } from 'react-toastify'
-
-import {
-  Popover,
-  Button,
-  PopoverInteractionKind,
-  Position,
-  ButtonGroup,
-  Alignment,
-  Intent,
-  Callout
-} from '@blueprintjs/core'
 
-import { Downloader } from '~/Pages/Components/Downloader'
+interface Props extends RouteComponentProps {
+  bots: any
+  workspace: any
+  fetchBots: any
+  fetchLicensing: any
+  licensing: any
+}
 
-class Bots extends Component {
+class Bots extends Component<Props> {
   state = {
     isCreateBotModalOpen: false,
+    isRollbackModalOpen: false,
+    isImportBotModalOpen: false,
     focusedBot: null,
-    isRollbackModalOpen: false
-  }
-
-  renderLoading() {
-    return <LoadingSection />
+    archiveUrl: undefined,
+    archiveName: ''
   }
 
   componentDidMount() {
-    this.downloadLink = React.createRef()
+    //  console.log(isOperationAllowed({ resource: 'admin.bots', operation: 'read' }))
```

Comment:
leftover

note that the package `debug` works in the browser too (`localStorage.debug = 'bp:*'`)